### DISCUSSION TITLE
MaterialVectorPostprocesor issue with stress component

### DISCUSSION THREAD

**Discussion by jessecarterMOOSE** [idaholab/moose#20834] [Q&A Modules: Solid mechanics] [answered]
If I am generating materials from the Tensor Mechanics Master Action, e.g. `generate_output = 'stress_yy'`, I can't seem to get them to work with `MaterialVectorPostprocessor`. It's odd because I can get them to work with a regular Postprocesor, like `ElementAvergeMaterialProperty`, but not the VPP.

For instance, this input file will work, but if I enable that `stress_yy_qps` MaterialVectorPostprocessor that operates on stress_yy, I am met with the error: `Unable to locate active object: stress_yy`

Update: If I create a new material with `RankTwoCartesianComponent` to grab that particular stress component, I can use the VPP on that new material, but still not the one generated by the Tensor Mechanics action. I don't understand why I need the intermediate step for the VPP.

```
[GlobalParams]
  displacements = 'disp_x disp_y'
[]

[Mesh]
  [square]
    type = GeneratedMeshGenerator
    dim = 2
    nx = 2
    ny = 2
  []
[]

[Modules/TensorMechanics/Master]
  [all]
    strain = SMALL
    planar_formulation = PLANE_STRAIN
    add_variables = true
    generate_output = 'stress_xx stress_xy stress_yy stress_zz strain_xx strain_xy strain_yy strain_zz'
  []
[]

[Functions]
  [pull]
    type = ParsedFunction
    value ='0.01 * t'
  []
[]

[BCs]
  [rightx]
    type = DirichletBC
    boundary = 1
    variable = disp_x
    value = 0.0
  []
  [bottomy]
    type = DirichletBC
    boundary = 0
    variable = disp_y
    value = 0.0
  []
  [pull]
    type = FunctionDirichletBC
    boundary = 2
    variable = disp_y
    function = pull
  []
[]

[Materials]
  [linear_stress]
    type = ComputeLinearElasticStress
    block = 0
  []
  [elasticity_tensor]
    type = ComputeIsotropicElasticityTensor
    block = 0
    poissons_ratio = 0.3
    youngs_modulus = 1e10
  []
  [foo]
    type = GenericConstantMaterial
    prop_names = foo
    prop_values = 42
  []
[]

[Executioner]
  type = Transient

  solve_type = PJFNK
  line_search = none

  l_max_its = 100
  l_tol = 1e-10
  nl_max_its = 15
  nl_rel_tol = 1e-12

  start_time = 0.0
  dt = 1.0
  dtmin = 1.0
  end_time = 5.0
[]

[Postprocessors]
  [avg_stress_yy]
    type = ElementAverageMaterialProperty
    mat_prop = stress_yy
  []
  [avg_foo]
    type = ElementAverageMaterialProperty
    mat_prop = foo
  []
[]

[VectorPostprocessors]
  inactive = 'stress_yy_qps'
  [stress_yy_qps]
    type = MaterialVectorPostprocessor
    material = stress_yy
    elem_ids = '1 2'
  []
  [foo_qps]
    type = MaterialVectorPostprocessor
    material = foo
    elem_ids = '1 2'
  []
[]

[Outputs]
  csv = true
  exodus = true
[]

```

**Comment by GiudGiud** [comment, marked as answer]
My hunch is that this has to do with the ordering of actions. 
VPPs might be created before the part of the TensorMechanicsAction that adds the material properties is ran. 
You can add 
```
[Debug]
  show_actions = true
[]
```
to see if this is true

**Reply by jessecarterMOOSE** [reply]
OK I think I figured it out. Based on your suggestion, I also had it print out the material properties. Looks like it's a naming problem.

Debug material property output looks like this:
```
      Material Name:       stress_yy_all
      Property Names:      "stress_yy" 
```
The MaterialVectorPostprocessor uses `getMaterialByName()` rather than `getMaterialProperty()` (I'm used to seeing the latter). So if I use `stress_yy_all`, it works. You wouldn't know it was called that unless you use the debug output. Seems like a bug?

**Reply by GiudGiud** [reply]
@dschwen if you want to change that
